### ISSUE TITLE
parseDate on functional bind (working with dates example) doesn't work / documentation confusing

### ISSUE THREAD

**Issue by LaughingBubba** [mskocik/svelty-picker#185]
I pretty much lifted the code from the [working-with-dates#example](https://svelty-picker.vercel.app/working-with-dates#example) and discovered the `parseDate` doesn't work or is deprecated. Then I realised the [functional example](https://svelte.dev/playground/1ddd82f573b94201b3c8fcab33bf0a46?version=5.9.0) referenced in the [migration guide](https://svelty-picker.vercel.app/migration-guide#initialdate) did it differently and more coherently. 

Just calling this out as I wasted a fair bit of time on this.

This doesn't work ====> 
```svelte
<script>
  ...
	
  let date = $state(( () => {
		let initDate = new Date()
		initDate.setHours(0,0,0)
		return initDate
	})() )

</script>

    <SveltyPicker
      inputId = "eventdatetime"
      inputClasses = ""
      mode = "date"
      format = "yyyy/mm/dd hh:ii"
      weekStart = {2}
      bind:value ={ () => {
                      return formatDate(date, 'yyyy/mm/dd', en, 'standard')
                    },
                    (v) => {
                      date = v ? parseDate(v, 'yyyy/mm/dd', en, 'standard') : null
                      console.log('raw "v" value', v)
                      console.log("parsed date", (v ? parseDate(v, 'yyyy/mm/dd', en, 'standard') : null))
                      console.log("state date", date)
                    }
                  }
    />
```

Produces:
![image](https://github.com/user-attachments/assets/6853af7a-dbd2-4eda-9e9c-d4f882cf5b7e)

But this does ====> 
```svelte
<script>
  ...

  let date = $state(new Date())

</script>

    <SveltyPicker
      inputId = "eventdatetime"
      inputClasses = ""
      mode = "date"
      format = "yyyy/mm/dd hh:ii"
      weekStart = {2}
      bind:value ={ () => {
                      return formatDate(date, 'yyyy/mm/dd', en, 'standard')
                    },
                    (v) => {
                      date = new Date(v)
                    }
                  }
    />
```

**Comment by mskocik** [comment]
> I pretty much lifted the code from the [working-with-dates#example](https://svelty-picker.vercel.app/working-with-dates#example)

Is something wrong/not working with example itself? Works for me 🤔 

> This doesn't work ====>

That's because you are parsing date in _date-only_ format, but you are providing date AND time part.
`yyyy/mm/dd` is NOT the same as `yyyy/mm/dd hh:ii`. Also I noticed, you are specifying `mode` as `date`, but providing `datetime` format. It is not compatible, and if date fails to parse/format, it returns "now" date as fallback value. Which is your case.

For you to simplify things, remove `mode` property. It's resolved automatically based on specified `format`. 

---

To summarize this, `parseDate` and `formatDate` works as intended, they are used in library itself. But you need to stick to exactly the same `format` value when parsing/formatting date, otherwise can get invalid date.
